### DISCUSSION TITLE
Maximum number of projects that I can create?

### DISCUSSION THREAD

**Discussion by clcl0811** [goharbor/harbor#15418] [Q&A]
Hi harbor community, I am curious about the guideline for creating projects in harbor:

- Is creating projects considered as low-cost to harbor?
- Is there a limitation (or, maximum number) regarding how many projects that harbor can host?
- If I create hundreds of projects or even thousands of projects, will that have negative impact on the harbor performance overall? (I understand that API calls for listing projects might be slower and that's expected)

(I tried to search for the answers by using Google and within this repo but didn't find anything)

Thank you.

**Comment by lookuptable** [comment]
Friendly ping.

**Comment by wy65701436** [comment]
Is creating projects considered as low-cost to harbor? 
Yes, it can be considered as low-cost as just DB data creation.

Is there a limitation (or, maximum number) regarding how many projects that harbor can host?
Actually, we didn't test that. But, In theory, it should be the maximum ID of postgres integer.

If I create hundreds of projects or even thousands of projects, will that have negative impact on the harbor performance overall? (I understand that API calls for listing projects might be slower and that's expected)
The performance issues that we see are highly related with concurrency number, not the project number.
